obspy/io/sac/tests/test_sacpz.py

```diff
@@ -63,6 +64,18 @@ def test_write_sacpz_multiple_channels(self):
         expected = [l for l in expected.split("\n") if "CREATED" not in l]
         self.assertEqual(got, expected)
 
+    def test_write_sacpz_soh(self):
+        path = os.path.join(self.path, '..', '..', 'stationxml', 'tests',
+                            'data', 'only_soh.xml')
+        inv = read_inventory(path)
+        f = io.StringIO()
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter('always')
+            inv.write(f, format='SACPZ')
+            self.assertEqual(len(w), 2)
```

Comment:
Ideally, this should also check the warning message, you might always get some warnings that are something else entirely.

I'm testing warnings like this nowadays: https://github.com/obspy/obspy/blob/06906f389bdaeb5fa4c4bd0cab9f066082e7da42/obspy/io/shapefile/tests/test_core.py#L179-L192

Ideally we should put this in a helper routine in `obspy/core/util/testing.py`..

```python
def assert_warning(warnings, message_regex, category=UserWarning):
    ...
```

Reply:
Py3 has `assertWarns` (which I originally implemented) and `assertWarnsRegex`, but not py2.
Although this may be remedied with the backport `import unittest2`, if we want to go that route.

I cringe at copy/pasting something from code to tests.
Catching exactly 2 warnings from the test file that has 2 warning conditions seemed sufficient. 
I can assert they are UserWarnings, or they end in `'. Skipping.'` or a full regex.
Anyway, thanks for caring about the code base. I'll proceed as advised.